### PR TITLE
98 add product delay metrics

### PR DESCRIPTION
**Pull request by mpejcoch**

### CODE REVIEW

File: internal/server/metrics.go [hunk 1]
```diff
@@ -31,14 +31,17 @@ type MetricsOpts struct {
...
 
 type metrics struct {
-	registry *prometheus.Registry
-	counter  *prometheus.CounterVec
-	duration *prometheus.HistogramVec
+	registry      *prometheus.Registry
+	counter       *prometheus.CounterVec
+	duration      *prometheus.HistogramVec
+	UptimeCounter *prometheus.Counter
 }
```
Comment (vkbo) [inline internal/server/metrics.go:38]: Inconsistent use of upper/lower case

File: internal/server/productstatus.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+type Product struct {
+	Name                 string
+	NextInstanceExpected time.Time
+}
+
+type Productstatus struct {
+	Products map[string]Product
+	gaugeVec *prometheus.GaugeVec
+}
```
Comment (vkbo) [inline internal/server/productstatus.go:19]: Inconsistent use of upper/lower case

File: internal/server/productstatus_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package server
+
+import (
+	"testing"
+	"time"
+
+	"github.com/metno/go-mms/pkg/mms"
+)
+
+func Test_pushEvent(t *testing.T) {
```
Comment (vkbo) [inline internal/server/productstatus_test.go:10]: Remove the underscore?

File: internal/server/productstatus.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+func NewProductstatus(m *metrics) *Productstatus {
```
Comment (vkbo) [inline internal/server/productstatus.go:21]: Should be NewProductStatus not NewProductstatus

File: internal/server/metrics.go (hunk 1 above)
Comment (vkbo) [reply internal/server/metrics.go:38]: I think they should all starte with an upper case letter

Comment (mpejcoch) [reply internal/server/metrics.go]: I forgot the Counter here, it's not being used, so I'm just dropping it.

File: internal/server/productstatus.go (hunk 4 above)
Comment (mpejcoch) [reply internal/server/productstatus.go:21]: It is a _Productstatus_ component here, which keeps track of multiple products, not a product status.

**Review by johtoblan — APPROVED** [review]

**Review by vkbo — APPROVED** [review]

**Pull request merged**